src/SiteAlias/ProcessManager.php

```diff
@@ -35,17 +35,21 @@ public function drush(SiteAliasInterface $siteAlias, string $command, array $arg
      */
     public function drushSiteProcess(SiteAliasInterface $siteAlias, array $args = [], array $options = [], array $options_double_dash = []): ProcessBase
     {
+        $drushScript = $this->drushScript($siteAlias);
+
         // Fill in the root and URI from the site alias, if the caller
         // did not already provide them in $options.
         if ($siteAlias->has('uri')) {
             $options += [ 'uri' => $siteAlias->uri(), ];
         }
-        if ($siteAlias->hasRoot()) {
+        // Include the --root parameter only if calling a global Drush.
+        // This is deprecated in Drush 12, and will be removed in Drush 13.
+        if ($siteAlias->hasRoot() && ($drushScript == "drush")) {
```

Comment:
> Is there any simplification we can do for local Drush calls?

If your site alias for your local Drupal site has a `--root` entry, then `$drush_script` will be an absolute path (to something relative to the root, or using `$_composer_bin_dir`). If your site alias for your local Drupal site does not have a `--root` entry, then we assume it's a multisite of the same Drupal install, and use the absolute path to the current running instance of Drush. Either way, the `--root` option is not passed along in the `exec`. The only way you get a `--root` here is if your alias has a `root` option and also defines `paths.drush_script` to be just "drush", i.e. the Drush in the $PATH. This is the one way that you could get Drush 12 to call a global Drush (earlier I incorrectly said there would be no way to do that).

The other way you can get simply "drush" here is if you're making a remote call using an alias that does not have a `--root` entry, e.g. a Pantheon alias. This will also call the global Drush. On Pantheon, the `drush` in the $PATH is `vendor/bin/drush`, as there is only one Drupal site in the container. I don't know that any other ISPs use this pattern. Pantheon used to have roots that looked like `/srv/binding/AN_ID_THAT_CAN_CHANGE/code/web`, but now it's always just `/code/web`, so we could start writing `root` entries into Pantheon aliases. Most folks are going to use `terminus drush` rather than Pantheon aliases, though, so this is something of an edge case.

Reply:
I was more thinking that in a local drush call like `drush --root=foo version`, the `--root` has no impact or should have no impact because the Drupal root is determined via the location of `drupal/core`.

Reply:
This code usually won't include the `--root` parameter. It will only be included if you also set `paths.drush-script` to "drush", which most current site aliases won't do, because that's the default in previous versions of Drush. So `--root` should go away in most instances.